src/main/java/org/mitre/synthea/modules/WeightLossModule.java

```diff
@@ -135,65 +132,35 @@ public void manageYearTwoThroughFive(Person person, long time) {
     boolean followsPlan = (boolean) person.attributes.get(WEIGHT_LOSS_ADHERENCE);
     boolean longTermSuccess = (boolean) person.attributes.get(LONG_TERM_WEIGHT_LOSS);
     int age = person.ageInYears(time);
-    int ageInMonths = person.ageInMonths(time);
     String gender = (String) person.attributes.get(Person.GENDER);
     // In the next 5 years, if someone has lost weight, check to see if they
     // will have long term success. If they don't, revert their weight back
-    // to the original weight
+    // to the original weight or BMI percentile
     if (followsPlan) {
       if (longTermSuccess) {
         if (age < 20) {
           // The person will continue to grow, increase their weight, but keep BMI steady
-          double height = lookupGrowthChart("height", gender, ageInMonths,
-              person.getVitalSign(VitalSign.HEIGHT_PERCENTILE, time));
-          double weight = maintainBMIPercentile(person, time);
-          person.setVitalSign(VitalSign.WEIGHT, weight);
-          person.setVitalSign(VitalSign.HEIGHT, height);
-          person.setVitalSign(VitalSign.BMI, bmi(height, weight));
-          setCurrentWeightForLengthPercentile(person, time);
+          maintainBMIPercentile(person, time);
         }
       } else {
-        double weight;
-        double height;
         if (age < 20) {
-          height = lookupGrowthChart("height", gender, ageInMonths,
-              person.getVitalSign(VitalSign.HEIGHT_PERCENTILE, time));
-
-          weight = pediatricRegression(person, time);
+          pediatricRegression(person, time);
         } else {
+          double weight;
+          double height;
           long start = (long) person.attributes.get(WEIGHT_MANAGEMENT_START);
           if (person.ageInYears(start) < 20) {
-            if (age >= 20) {
-              height = lookupGrowthChart("height", gender, 240,
-                  person.getVitalSign(VitalSign.HEIGHT_PERCENTILE, time));
-            } else {
-              height = lookupGrowthChart("height", gender, ageInMonths,
-                  person.getVitalSign(VitalSign.HEIGHT_PERCENTILE, time));
-            }
-
+            height = growthChart.get(GrowthChart.ChartType.HEIGHT).lookUp(240, gender,
```

Comment:
If they are less than 20, lookup their height as if they were 240 months old?

Reply:
I probably need to checkout the code... so many changes it is hard to follow without looking at it cleanly.

Reply:
Not exactly.

This branch is to deal with the case where you have someone who lost weight before they were 20, but are now regressing, but their regression will cross age 20. It's kind of a pain to deal with because we don't bother with height changes after that in Synthea, but prior to that, height can change, which will impact BMI.

This code is essentially saying that if you started before you were 20, but you are 20 or older now, use the value for height for 20 year olds, regardless of how far into their twenties the individual is.